lib/gdpr/gateway/user_details.rb

```diff
@@ -79,12 +71,36 @@ def notify_inactive_users
       if user.mobile?
         WifiUser::SMSSender.send_credentials_expiring_notification(username, contact)
       elsif user.valid_email?(contact)
-        WifiUser::EmailSender.send_credentials_expiring_notification(username, contact)
+        begin
+          WifiUser::EmailSender.send_credentials_expiring_notification(username, contact)
+          @logger.info("Email sent to #{username} at #{contact}")
+        rescue Notifications::Client::BadRequestError => e
+          handle_email_error(e, username, contact)
+        end
       else
-        logger.warn("Invalid email address for user #{username}: #{contact}")
+        @logger.warn("Invalid contact for user #{username}")
       end
     end
 
-    logger.info("Finished notification process for users who were inactive for 11 months")
+    @logger.info("Finished notification process for users inactive for 11 months")
+  end
+
+private
+
+  def find_inactive_users(months)
```

Comment:
Does this cause inactive users being sent this email every day for a month? (assuming this runs daily)

Reply:
Yes, but it takes a month before the inactive users are actually deleted right? So between 11 and 12 months of inactivity do users not get emailed every day?